rmgpy/data/kinetics/family.py

```diff
@@ -1873,11 +1873,16 @@ def __generateReactions(self, reactants, products=None, forward=True, prod_reson
 
         if forward:
             template = self.forwardTemplate
+            reactantNum = self.reactantNum
         elif self.reverseTemplate is None:
             return []
         else:
             template = self.reverseTemplate
-
+            reactantNum = self.productNum
+        
+        if self.autoGenerated and reactantNum != len(reactants):
```

Comment:
Does it have to be autoGenerated for this to be a good check to do?

Reply:
Yes, because for non-autogenerated trees self.reactantNum seems to be guessed from the template number, which really only works for bimolecular families.